forge-game/src/main/java/forge/game/trigger/TriggerHandler.java

```diff
@@ -266,7 +266,7 @@ public final void runTrigger(final TriggerType mode, final Map<AbilityKey, Objec
 
         if (mode == TriggerType.Always) {
             runStateTrigger(runParams);
-        } else if (game.getStack().isFrozen() || holdTrigger) {
+        } else if ((game.getStack().isFrozen() || holdTrigger) && mode != TriggerType.TapsForMana && mode != TriggerType.ManaAdded) {
```

Comment:
Oh i see, we're not even getting to MagicStack before they get held.  We need to test if the trigger is a mana ability here to run it right away. We can merge for now and i'll try to clean it up soon.